### PR TITLE
add fail condition for invalid perf rawevents and metric events

### PR DESCRIPTION
**Pull request by disgoel**
patch1 adds fail condition for invalid metric events
fix error as process.getstatusoutput has no attribute stdout/stderr
changed perf stat metric events to run with cpu 0

patch2 adds fail condition for invalid perf_rawevents

Signed-off-by: Disha Goel <disgoel@linux.ibm.com>

avocado run --test-runner runner perf_rawevents.py
JOB ID     : 050a3688bba2b2fc14e16aa1532a5f9638480c07
JOB LOG    : /home/disha/avocado-fvt-wrapper/results/job-2023-05-10T18.54-050a368/job.log
 (1/2) perf_rawevents.py:PerfRawevents.test_raw_code: FAIL: perf_raw_events: refer log file for failed events (11.41 s)
 (2/2) perf_rawevents.py:PerfRawevents.test_name_event: PASS (13.63 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/disha/avocado-fvt-wrapper/results/job-2023-05-10T18.54-050a368/results.html
JOB TIME   : 33.78 s
[job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/11485888/job.log)

avocado run --test-runner runner perf_metric.py
JOB ID     : 8ca83c528ce357d046902a6cf1ddabf59b79aef9
JOB LOG    : /home/disha/avocado-fvt-wrapper/results/job-2023-05-16T14.47-8ca83c5/job.log
 (1/2) perf_metric.py:perf_metric.test_all_metric_events_with_M: PASS (191.68 s)
 (2/2) perf_metric.py:perf_metric.test_all_metric_events_with_metric: PASS (191.57 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/disha/avocado-fvt-wrapper/results/job-2023-05-16T14.47-8ca83c5/results.html
JOB TIME   : 391.84 s
[job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/11486148/job.log)

### CODE REVIEW

File: perf/perf_metric.py [hunk 1]
```diff
@@ -82,12 +82,11 @@ def setUp(self):
 
     def _run_cmd(self, option):
         for line in self.list_of_metric_events:
-            cmd = "perf stat %s %s sleep 1" % (option, line)
-            rc, op = process.getstatusoutput(cmd, ignore_status=True,
-                                             shell=True, verbose=True)
+            cmd = "perf stat %s %s -C 0 sleep 1" % (option, line)
+            op = process.run(cmd, ignore_status=True, shell=True, verbose=True)
+            output = op.stdout.decode() + op.stderr.decode()
```
Comment (abdhaleegit) [inline perf/perf_metric.py:87]: cant I directly decode the whole output 
```
op = process.run(cmd, ignore_status=True, shell=True, verbose=True).decode()
```

File: perf/perf_rawevents.py [hunk 2]
```diff
@@ -78,9 +78,10 @@ def setUp(self):
     def run_event(self, filename, perf_flags):
         for line in genio.read_all_lines(filename):
             cmd = "%s%s sleep 1" % (perf_flags, line)
-            output = process.run(cmd, shell=True,
-                                 ignore_status=True)
-            if output.exit_status != 0:
+            result = process.run(cmd, shell=True, ignore_status=True)
```
Comment (abdhaleegit) [inline perf/perf_rawevents.py:81]: same here  result = process.run(cmd, shell=True, ignore_status=True).decode()

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

File: perf/perf_metric.py (hunk 1 above)
Comment (disgoel) [reply perf/perf_metric.py:87]: @abdhaleegit its not possible to directly decode the output
`AttributeError: 'CmdResult' object has no attribute 'decode'`

Comment (abdhaleegit) [reply perf/perf_metric.py:87]: ok it works only for output =  process.system_output().decode()   and not with process.run()

Comment (disgoel) [reply perf/perf_metric.py:87]: correct and we need here process.run()

Comment (abdhaleegit) [reply perf/perf_metric.py:87]: we are good than

Comment (abdhaleegit) [inline perf/perf_metric.py:87]: make use of % operator to concatenate and also avoid  redundant decode

output = ('%s %s' % (op.stdout, op.stderr)).decode()

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
Rest all LGTM

File: perf/perf_metric.py (hunk 1 above)
Comment (disgoel) [reply perf/perf_metric.py:87]: The %s formatting operator performs string interpolation, converting the byte strings to string representations and concatenating them, so we cant use decode() with it.
```
>>> output = ('%s %s' % (op.stdout, op.stderr)).decode()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'str' object has no attribute 'decode'
```
We can use either of this two
```
output = '%s %s' % (op.stdout.decode(), op.stderr.decode())
output = (op.stdout + op.stderr).decode()
```

Comment (abdhaleegit) [reply perf/perf_metric.py:87]: Alright, Please go with output = (op.stdout + op.stderr).decode()

Comment (disgoel) [reply perf/perf_metric.py]: I have done the changes and pushed.

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Pull request merged**